### PR TITLE
Allow customization of Github API endpoint

### PR DESCRIPTION
**Pull request by maallen**
Allows a custom Github endpoint to be set from configuration, if no configuration is set it defaults to `https://api.github.com`

### CODE REVIEW

**Review by garionpin — APPROVED** [review]

**Review by ehoogerbeets — COMMENTED** [review]
Thanks for putting that in!

File: common/src/main/java/com/box/l10n/mojito/github/GithubClient.java [hunk 1]
```diff
@@ -57,10 +59,12 @@ public GithubClient(String appId, String key, String owner, long tokenTTL) {
     }
     this.owner = owner;
     this.tokenTTL = tokenTTL;
+    this.endpoint =
+        endpoint.endsWith("/") ? endpoint.substring(0, endpoint.length() - 1) : endpoint;
   }
 
   public GithubClient(String appId, String key, String owner) {
-    this(appId, key, owner, 60000L);
+    this(appId, key, owner, 60000L, "https://api.github.com");
```
Comment (aurambaj) [inline common/src/main/java/com/box/l10n/mojito/github/GithubClient.java:67]: can we grab that from their constant "GITHUB_URL"?

File: common/src/main/java/com/box/l10n/mojito/github/GithubClientConfiguration.java [hunk 2]
```diff
@@ -10,6 +10,8 @@ public class GithubClientConfiguration {
 
   Long tokenTTL = 60000L;
 
+  String endpoint = "https://api.github.com";
```
Comment (aurambaj) [inline common/src/main/java/com/box/l10n/mojito/github/GithubClientConfiguration.java:13]: same?

File: common/src/main/java/com/box/l10n/mojito/github/GithubClient.java (hunk 1 above)
Comment (maallen) [reply common/src/main/java/com/box/l10n/mojito/github/GithubClient.java:67]: No, it's not a public constant so can't be accessed outside of the `org.kohsuke.github` package

**Review by aurambaj — APPROVED** [review]

**Pull request merged**